### PR TITLE
Infinispan quickstart with docker compose

### LINKED ISSUE #148: infinispan-client quickstart can't run using docker - server case
infinispan-client quickstart can't run using docker (both client and server cases)

Instructions in README suggest to run ISPN server in docker or on the main system.
Running server in docker causes stuck client application, I had to `kill -9` it.
I think it's because of the docker nature where the server is on different network than client.

Errors from client when running against ISPN server in docker:
```
2019-04-12 09:08:19,549 INFO  [org.inf.cli.hot.imp.tra.net.ChannelFactory] (HotRod-client-async-pool-0)
   ISPN004014: New server added(172.17.0.3:11222), adding to the pool.
2019-04-12 09:08:19,549 INFO  [org.inf.cli.hot.imp.tra.net.ChannelFactory] (HotRod-client-async-pool-0)
   ISPN004016: Server not in cluster anymore(localhost:11222), removing from the pool.
2019-04-12 09:09:19,548

  WARN  [org.inf.cli.hot.imp.tra.net.ChannelFactory] (HotRod-client-async-pool-1)
  ISPN004015: Failed adding new server 172.17.0.3:11222: io.netty.channel.ConnectTimeoutException:
  connection timed out: /172.17.0.3:11222
```

### PR DESCRIPTION
**Pull request by karesti**
Using docker-compose to run the client and infinispan-server using docker

closes #148

### CODE REVIEW

**Comment by karesti** [comment]
@rsvoboda I tried using "wait for it" but it did not "wait for it" enough (could not make it work!! 🙈) so I created two docker-compose so one to be run after the other. Not sure this is the best but still something to work on. the server list is now overrideable at runtime so this is the good time for this PR 💃

File: infinispan-client/README.md [hunk 1]
```diff
@@ -25,9 +25,24 @@ with the following property:
...
+
+Once you built a docker image using the `Dockerfile.native`, you might want to test this
+container connecting to a running Infinispan image.
+
+Infinispan needs to be properly started to test this locally, and the containers must be in the same network.
+
+For that, we have provided 2 docker-compose files. 
+
+Run and wait for start `docker-compose -f infinispan-docker-compose.yaml up`
+Then run `docker-compose -f infinispan-docker-compose.yaml up`
```
Comment (rsvoboda) [inline infinispan-client/README.md]: `docker-compose -f client-docker-compose.yaml up`

File: infinispan-client/infinispan-docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+services:
+  infinispan-server:
+    image: jboss/infinispan-server:10.0.0.Beta3
+    ports:
+    - "11222:11222"
+    container_name: infinispan
+    networks:
+      - mynetwork
+networks:
```
Comment (rsvoboda) [inline infinispan-client/infinispan-docker-compose.yaml]: Please move `networks` before `services` as in client-docker-compose.yaml so both files have same format

**Comment by rsvoboda** [comment]
So I played a bit with docker compose files and have two ways how to merge those 2 files:

**Option A**
```yaml
version: '3'

networks:
  mynetwork:
    driver: bridge

services:
  infinispan-server:
    image: jboss/infinispan-server:10.0.0.Beta3
    ports:
    - "11222:11222"
    container_name: infinispan
    networks:
      - mynetwork
  client-app:
    image: quarkus/infinispan-client
    ports:
      - "8081:8081"
    container_name: client
    depends_on:
      - infinispan-server
    restart: on-failure
    networks:
      - mynetwork
```
Nicer file, but `restart: on-failure` causes several client app restarts (and stacktraces in log) before server is available => worse user experience

**Option B**
```yaml
version: '3'

networks:
  mynetwork:
    driver: bridge

services:
  infinispan-server:
    image: jboss/infinispan-server:10.0.0.Beta3
    environment:
      - APP_USER=user
      - APP_PASS=changeme
    ports:
      - "11222:11222"
    container_name: infinispan
    networks:
      - mynetwork
  client-app:
    image: quarkus/infinispan-client
    ports:
      - "8081:8081"
    container_name: client
    depends_on:
      - infinispan-server
    networks:
      - mynetwork
    command: >
      /bin/sh -c "
        echo Waiting for Infinispan service start ...;
        while ! curl --silent --output /dev/null --fail-early --user user:changeme http://infinispan:8080;
        do
          sleep 1;
        done;
        echo Infinispan service started!;
        ./application -Dquarkus.http.host=0.0.0.0 -Dquarkus.infinispan-client.server-list=infinispan:11222
      "
```
Longer file, nicer end user experience as the log is not polluted by exception stacktraces.
Also application start command has to be here explicitly defined.

@karesti what's your preference ? Any other idea ?

**Comment by karesti** [comment]
@rsvoboda My choice goes for the longer file with a better user experience. If we document that we are just "waiting for the server" and that is is something needed for testing purposes I think it's totally fine!

**Comment by karesti** [comment]
@rsvoboda thanks for the help on the docker-compose!! I pushed the change

**Pull request merged**